### PR TITLE
fix: issues with quick transition between LOD - Scene - LOD - Scene

### PR DESCRIPTION
**Pull request by fcolarich**
## What does this PR change?

Added a small time before we switch from a scene/scene promise to a LOD, reducing the visual glitches and reducing the edge cases caused by quickly panning around the camera.

## How to test the changes?

Go to -150,0, zoom into 1st person view, then approach the lonely house at -150,-3 until it changes from LOD to Scene. Then zoom out. It should take about 5seconds until it turns into LOD again. If you zoom in, it will turn into a scene again.
This delay should work for all scenes allowing to pan the camera around and leave a scene and come back without it changing too quickly to LODs.
Please let me know if the time set is OK or if we need more time for the change to happen.

https://github.com/user-attachments/assets/a8cb663d-37d6-49f0-ac6d-8b21bdd174cc

### CODE REVIEW

**Review by anicalbano — COMMENTED** [review]
🟢 Checked and working as expected! Alongside the test, the following smoke was performed:

✔️ Backpack/Emotes/Wearables
✔️ Social interaction on Chat/My Profile
✔️ Teleport with /goto and Map

https://github.com/user-attachments/assets/3b0f5eec-a1d7-4d50-aabf-2f3a84e7675e

Leaving a commend and awaiting for Mac test for fully approval 😄

**Review by DafGreco — APPROVED** [review]
Implementation done on macOS ✅ 
- Backpack
- Scenes 
-Teleport

**Review by anicalbano — APPROVED** [review]

File: Explorer/Assets/DCL/LOD/Systems/UpdateVisualSceneStateSystem.cs [hunk 1]
```diff
@@ -107,20 +107,28 @@ private void IterateOverOneOf<T>(Archetype archetype, ContinuationMethod<T> cont
...
 
-                        // we call it directly so we avoid an extra query
-                        if (visualSceneStateComponent.IsDirty)
-                            continuationMethod(entity, ref visualSceneStateComponent, ref sceneDefinitionComponent, ref partitionComponent, ref customComponent);
-                    }
+                    // we call it directly so we avoid an extra query
+                    if (visualSceneStateComponent.IsDirty)
+                        continuationMethod(entity, deltaTime, ref visualSceneStateComponent, ref sceneDefinitionComponent, ref partitionComponent, ref customComponent);
+                    else
+                        visualSceneStateComponent.TimeToChange = 0;
```
Comment (dalkia) [inline Explorer/Assets/DCL/LOD/Systems/UpdateVisualSceneStateSystem.cs:116]: TO make it clear, what do you think if we reset the time when we mark it as dirty?

File: Explorer/Assets/DCL/LOD/Systems/VisualSceneStateResolver.cs [hunk 2]
```diff
@@ -20,23 +20,22 @@ public void ResolveVisualSceneState(ref VisualSceneState visualSceneState, Parti
...
+            else if (!sceneDefinitionComponent.IsSDK7) visualSceneState.CandidateVisualSceneState = visualSceneState.CurrentVisualSceneState = VisualSceneStateEnum.SHOWING_LOD;
             else
             {
-                var candidateState = partition.Bucket < lodSettingsAsset.SDK7LodThreshold
+                visualSceneState.CandidateVisualSceneState = partition.Bucket < lodSettingsAsset.SDK7LodThreshold
                     ? VisualSceneStateEnum.SHOWING_SCENE
                     : VisualSceneStateEnum.SHOWING_LOD;
 
-                if (candidateState != visualSceneState.CurrentVisualSceneState)
+                if (visualSceneState.CandidateVisualSceneState != visualSceneState.CurrentVisualSceneState)
```
Comment (dalkia) [inline Explorer/Assets/DCL/LOD/Systems/VisualSceneStateResolver.cs:37]: I mean, reset the time here. I think it makes more sense

**Review by dalkia — APPROVED** [review]
Nice work 💪 . Left a comment to improve clarity in my opinion

File: Explorer/Assets/DCL/LOD/Systems/UpdateVisualSceneStateSystem.cs (hunk 1 above)
Comment (fcolarich) [reply Explorer/Assets/DCL/LOD/Systems/UpdateVisualSceneStateSystem.cs:116]: sounds good :)

**Review by pravusjif — APPROVED** [review]

File: Explorer/Assets/DCL/LOD/Settings/ILODSettingsAsset.cs
Comment (QThund) [inline Explorer/Assets/DCL/LOD/Settings/ILODSettingsAsset.cs:1]: Please write documentation to provide some context about this new field, why is it necessary, what is its purpose or who uses it for what.

Comment (QThund) [inline Explorer/Assets/DCL/LOD/Settings/ILODSettingsAsset.cs:1]: Please write documentation to provide some context about this new field, why is it necessary, what is its purpose or who uses it for what.

File: Explorer/Assets/DCL/LOD/Settings/ILODSettingsAsset.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@ public interface ILODSettingsAsset
         //Threshold for bucket partition (inclusive)
         int[] LodPartitionBucketThresholds { get;  }
         int SDK7LodThreshold { get; set;  }
+        float TimeToChangeToLod { get; set; }
```
Comment (QThund) [inline Explorer/Assets/DCL/LOD/Settings/ILODSettingsAsset.cs:13]: Please write documentation to provide some context about this new field, why is it necessary, what is its purpose or who uses it for what.

File: Explorer/Assets/DCL/LOD/Settings/LODSettingsAsset.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ public class LODSettingsAsset : ScriptableObject, ILODSettingsAsset
         };
 
         [field: SerializeField] public int SDK7LodThreshold { get; set; } = 2;
+        [field: SerializeField] public float TimeToChangeToLod { get; set; } = 5;
```
Comment (QThund) [inline Explorer/Assets/DCL/LOD/Settings/LODSettingsAsset.cs:18]: Please write documentation or tooltip to provide some context about this new field, why is it necessary, what is its purpose or who uses it for what.

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Components/VisualSceneState.cs [hunk 5]
```diff
@@ -7,7 +7,9 @@ namespace ECS.SceneLifeCycle.Components
     public struct VisualSceneState
     {
         public VisualSceneStateEnum CurrentVisualSceneState;
+        public VisualSceneStateEnum CandidateVisualSceneState;
         public bool IsDirty;
+        public float TimeToChange;
```
Comment (QThund) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/Components/VisualSceneState.cs:12]: Please write documentation to provide some context about this new field, why is it necessary, what is its purpose or who uses it for what.

**Review by QThund — APPROVED** [review]
Approved! My comments are all the same, about providing documentation.
